src/theming/theming-decorator.ts

```diff
@@ -1,38 +1,71 @@
-import type { ReactiveElement } from 'lit';
+import type { LitElement } from 'lit';
 
-import { updateWhenThemeChanges } from './theming-controller.js';
+import { ThemingController } from './theming-controller.js';
 import type { Themes } from './types.js';
 
 /**
  * Class decorator to enable multiple theme support for a component.
  * The component will re-render on theme changes.
  *
- * See also {@link updateWhenThemeChanges} for the same functionality
- * without the use of this decorator.
+ * Passing `{ exposeController: true }` in the decorator options will create an internal symbol
+ * which can be used by {@link getThemeController} to additionally query and modify the component
+ * based on the themes.
  *
  * Usage:
  *  ```ts
  *  import { LitElement, html } from 'lit';
  *  import { themes } from 'igniteui-webcomponents/themes';
- *  import { styles } from './themes/my-element.base.css';
- *  import { styles as material } from './themes/my-element.material.css';
- *  import { styles as bootstrap } from './themes/my-element.bootstrap.css';
- *  import { styles as indigo } from './themes/my-element.indigo.css';
+ *  import { styles } from './themes/my-element.base.css.js';
+ *  import { styles as shared } from './themes/shared/my-picker.common.css.js';
+ *  import { all } from './themes/themes.js';
  *
- *  @themes({ material, bootstrap, indigo })
+ *  @themes(all)
  *  class MyElement extends LitElement {
- *    public static styles = styles;
+ *    public static styles = [styles, shared];
+ *  }
+ *
+ *  @themes(all, { exposeController: true })
+ *  class MyElementWithExposedTheming extends LitElement {
+ *    ...
+ *
+ *    render() {
+ *      const theme = getThemeController(this)?.theme;
+ *      ...
+ *    }
  *  }
  *  ```
  */
-export function themes(themes: Themes, exposeTheme = false) {
-  return (clazz: any) => {
-    clazz.addInitializer((instance: ReactiveElement) => {
-      updateWhenThemeChanges(instance, themes, exposeTheme);
-    });
+export function themes(themes: Themes, options?: ThemeOptions) {
+  return (proto: unknown) => {
+    (proto as typeof LitElement).addInitializer((instance) => {
+      const controller = new ThemingController(instance, themes);
+      instance.addController(controller);
 
-    return clazz;
+      if (options?.exposeController) {
+        Object.defineProperty(instance, themeSymbol, {
+          get() {
+            return controller;
+          },
+          configurable: true,
+          enumerable: false,
+        });
+      }
+    });
   };
 }
 
-export { themeSymbol } from './theming-controller.js';
+/**
+ * Returns the theming controller for the given element if exposed.
+ */
+export function getThemeController(host: LitElement) {
+  return isControllerExposed(host) ? host[themeSymbol] : undefined;
+}
+
+function isControllerExposed(
+  host: LitElement
+): host is LitElement & { [themeSymbol]: ThemingController } {
```

Comment:
@rkaraivanov Isn't this supposed to return the `ThemeController` interface? Otherwise that interface is goes pretty unused atm.